crates/turborepo-lib/src/config/override_env.rs

```diff
@@ -0,0 +1,493 @@
+                .reason("Vercel wins: disregard just TURBO_TEAMID")
+                .TURBO_TEAMID()
+                .VERCEL_ARTIFACTS_OWNER()
+                .VERCEL_ARTIFACTS_TOKEN()
+                .team_id(VERCEL_ARTIFACTS_OWNER.clone())
+                .token(VERCEL_ARTIFACTS_TOKEN.clone()),
+            TestCase::new()
+                .reason("Vercel wins if TURBO_TOKEN is missing")
+                .TURBO_TEAM()
+                .TURBO_TEAMID()
+                .VERCEL_ARTIFACTS_OWNER()
+                .VERCEL_ARTIFACTS_TOKEN()
+                .team_id(VERCEL_ARTIFACTS_OWNER.clone())
+                .token(VERCEL_ARTIFACTS_TOKEN.clone()),
+            //
+            // Just get a team_id
+            // ------------------------------
+            TestCase::new()
+                .reason("just VERCEL_ARTIFACTS_OWNER")
+                .VERCEL_ARTIFACTS_OWNER()
+                .team_id(VERCEL_ARTIFACTS_OWNER.clone()),
+            TestCase::new()
+                .reason("just TURBO_TEAMID")
+                .TURBO_TEAMID()
+                .team_id(TURBO_TEAMID.clone()),
+            //
+            // Just get a team_slug
+            // ------------------------------
+            TestCase::new()
+                .reason("just TURBO_TEAM")
+                .TURBO_TEAM()
+                .team_slug(TURBO_TEAM.clone()),
+            //
+            // just team_slug and team_id
+            // ------------------------------
+            TestCase::new()
+                .reason("if we just have TURBO_TEAM+TURBO_TEAMID, that's ok")
+                .TURBO_TEAM()
+                .TURBO_TEAMID()
+                .team_slug(TURBO_TEAM.clone())
+                .team_id(TURBO_TEAMID.clone()),
+            //
+            // just set team_id and team_slug
+            // ------------------------------
+            TestCase::new()
+                .reason("if we just have a TURBO_TEAM and TURBO_TEAMID we can use them both")
+                .TURBO_TEAM()
+                .TURBO_TEAMID()
+                .team_id(TURBO_TEAMID.clone())
+                .team_slug(TURBO_TEAM.clone()),
+        ];
+
+        for case in cases {
+            let mut env: HashMap<OsString, OsString> = HashMap::new();
+
+            if let Some(value) = &case.input.TURBO_TEAM {
+                env.insert("turbo_team".into(), value.into());
+            }
+            if let Some(value) = &case.input.TURBO_TEAMID {
+                env.insert("turbo_teamid".into(), value.into());
+            }
+            if let Some(value) = &case.input.TURBO_TOKEN {
+                env.insert("turbo_token".into(), value.into());
+            }
+            if let Some(value) = &case.input.VERCEL_ARTIFACTS_OWNER {
+                env.insert("vercel_artifacts_owner".into(), value.into());
+            }
+            if let Some(value) = &case.input.VERCEL_ARTIFACTS_TOKEN {
+                env.insert("vercel_artifacts_token".into(), value.into());
+            }
```

Comment:
It would be cool to also make sure that `Input::from` is producing the expected `Input`
```suggestion
            }
            
            let actual_input = Input::from(&env);
            assert_eq!(case.input, actual_input);
```

Reply:
wonderful! ac49766bb27dfa017d4074cdc9584b88094e70a7